src/backend/executor/nodeNestloop.c

```diff
@@ -585,6 +585,8 @@ ExecReScanNestLoop(NestLoopState *node)
 {
 	PlanState  *outerPlan = outerPlanState(node);
 
+	Assert(outerPlan);
```

Comment:
May be add comment before, like: NestLoopState always has outerPlanState

Reply:
I think it's obvious that outerPlan shouldn't be null just because it's dereferenced here

Reply:
Adding an assertion is not a solution to the problem. The static analyzer checks the release build.

Reply:
Then the only solution I see is replace assert with if (... == NULL) ereport

Reply:
Why report an error to the user if nothing can be done about it?

Reply:
You may use elog, it doesn't matter. My point is that the only way to stop the analyzer complining is to throw a runtime error at the veru place a variable was derefernced rather then adding these check in the places pointer be the analyzer